parlai/core/torch_generator_agent.py

```diff
@@ -359,15 +367,16 @@ def build_criterion(self):
         if self.use_cuda:
             self.criterion.cuda()
 
-    def _init_cuda_buffer(self, model, criterion, batchsize, maxlen):
+    def _init_cuda_buffer(self, batchsize, maxlen, force=False):
         """Pre-initialize CUDA buffer by doing fake forward pass."""
-        if self.use_cuda and not hasattr(self, 'buffer_initialized'):
+        if self.use_cuda and (force or not hasattr(self, 'buffer_initialized')):
             try:
-                print('preinitializing pytorch cuda buffer')
                 dummy_xs = torch.ones(batchsize, maxlen).long().cuda()
                 dummy_ys = torch.ones(batchsize, 2).long().cuda()
-                scores, _, _ = model(dummy_xs, dummy_ys)
-                loss = criterion(scores.view(-1, scores.size(-1)), dummy_ys.view(-1))
+                scores, _, _ = self.model(dummy_xs, dummy_ys)
```

Comment:
maybe a silly question, but can we assume that all future torch generator agents will have the same input/output format?

Reply:
I'm going with Cross That Bridge Later